### ISSUE TITLE
A11y: No label for calcite-tile-select-group in screen reader

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-components#2259]
### Summary

When using a `calcite-tile-select-group` the text present in the tile is not read by the screen reader.

### Actual Behavior

Screen reader reads "radio button, 1 of 4"

### Expected Behavior

Screen reader should announce "TEXT OF TILE, radio button, 1 of 4"

### Reproduction Steps

1. Go to calcite select group story
2. turn on voiceOver

### Relevant Info 

This presents in Apple VoiceOver, but assume NVDA or JAWS would also behave poorly.

### Suggestion

Add a hidden label component connected to the radio button with the text of the tile's heading

**Comment by driskull** [comment]
@paulcpederson Should the `calcite-label` be inside the component or should it wrap the `calcite-tile`? 

It would be nice if it used the `calcite-tile.heading`, however, it seems that the heading is optional and we would need it to be required if the tile has `input-enabled`.

**Comment by driskull** [comment]
This can be fixed by just adding an `aria-label` on the `radio/switch`.

![image](https://user-images.githubusercontent.com/1231455/122126360-77bd5100-cde6-11eb-9a33-6277b607b8f3.png)

**Comment by jcfranco** [comment]
Verified using https://esri.github.io/calcite-components/?path=/story/components-tiles-tile-select-group--light + VoicerOver.
